File updater checker no longer feezes file list

lzap opened the pull request:
FileUpdateChecker initializer modifies input parameter `files` by calling `freeze`. This can lead to unexpected behaviour when things are set to initialize in different than standard order in Rails. This patch makes sure FileUpdateChecker has it's own frozen copy of files to watch.



More details about the problems we have can be found at:

https://github.com/Shopify/bootsnap/issues/106#issuecomment-381254534

lzap committed:
File updater checker no longer feezes file list

kamipo commented on activesupport/lib/active_support/file_update_checker.rb:
```diff
@@ -44,7 +44,7 @@ def initialize(files, dirs = {}, &block)
       end
 
-      @files = files.freeze
+      @files = files.dup.freeze
```
The `freeze` was added at 5ea2b11ad7fb1e0e0580edd9d8e0e688cfffa3f2 to ensure the files will not be changed.

Generally passed argument should not be mutated, but https://github.com/Shopify/bootsnap/issues/106 is strange to me. If `can't modify frozen Array` is raised by the `freeze`, it means that the files are changed after FileUpdateChecker has been initialized, it should not be happenned for FileUpdateChecker. Isn't it?

lzap commented:
Hmm no luck, I removed that commit which we added for significant boost of startup time: https://github.com/theforeman/foreman/commit/c2e58b1b275e0107de4089e15adc8b6203e696b1

No change. One important point to mention is that this only happens in rake (tests), the application loads fine.
